app/src/main/java/com/codesoom/assignment/application/ProductService.java

```diff
@@ -1,64 +1,105 @@
 package com.codesoom.assignment.application;
 
-import com.codesoom.assignment.ProductNotFoundException;
+import com.codesoom.assignment.exception.ProductBadRequestException;
+import com.codesoom.assignment.exception.ProductNotFoundException;
 import com.codesoom.assignment.domain.Product;
 import com.codesoom.assignment.domain.ProductRepository;
 import com.codesoom.assignment.dto.ProductData;
+import com.github.dozermapper.core.DozerBeanMapperBuilder;
+import com.github.dozermapper.core.Mapper;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
 import javax.transaction.Transactional;
 import java.util.List;
 
+/**
+ * @description 상품 관련 CRUD
+ */
 @Service
 @Transactional
+@RequiredArgsConstructor
 public class ProductService {
     private final ProductRepository productRepository;
+    private final Mapper mapper = DozerBeanMapperBuilder.buildDefault();
 
-    public ProductService(ProductRepository productRepository) {
-        this.productRepository = productRepository;
-    }
-
-    public List<Product> getProducts() {
+    /**
+     * @description 전체 상품 목록을 리턴한다.
+     *
+     * @return 저장되어 있는 전체 상품 목록
+     */
+    public List<Product> findAll() {
         return productRepository.findAll();
     }
 
+    /**
+     * @description 식별자에 해당하는 상품을 리턴한다.
+     *
+     * @param id - 조회하고자 하는 상품의 식별자
+     * @return 주어진 식별자에 해당하는 상품
+     * @throws ProductNotFoundException
+     *          주어진 상품의 정보가 유효하지 않은 경우
+     */
     public Product getProduct(Long id) {
-        return findProduct(id);
+        return productRepository.findById(id)
+                .orElseThrow(() -> new ProductNotFoundException(id));
     }
 
+    /**
+     * @description 상품을 저장하고 해당 객체를 리턴한다.
+     *
+     * @param productData - 새로 저장하고자 하는 상품
+     * @return 저장 된 상품
+     * @throws ProductBadRequestException 만약 주어진 상품의
+     *         이름이 비어있거나, 메이커가 비어있거나, 가격이 비어있는 경우
+     */
     public Product createProduct(ProductData productData) {
-        Product product = Product.builder()
-                .name(productData.getName())
-                .maker(productData.getMaker())
-                .price(productData.getPrice())
-                .imageUrl(productData.getImageUrl())
-                .build();
+
+        if(productData.getName().isBlank())
+            throw new ProductBadRequestException("name");
+
+        if(productData.getMaker().isBlank())
+            throw new ProductBadRequestException("maker");
+
+        if(productData.getPrice() == null) {
+            throw new ProductBadRequestException("price");
+        }
+
+        Product product = mapper.map(productData,Product.class);
```

Comment:
```suggestion
        Product product = mapper.map(productData, Product.class);
```

빈칸 하나를 추가하면 좋을 것 같아요. 에디터의 자동 정렬 기능을 활용하시면 더 편할 것 같습니다